src/redux/slices/events/middleware.ts

```diff
@@ -0,0 +1,29 @@
+import API from '@axios/API'
+import { IError } from '@axios/authentication/authManagerTypes'
+import { AppDispatch } from '@redux/store'
+
+import slice from './slice'
+
+const { setEventsLoading, setError, setEventsData, setPageSize, setTotalItems } = slice.actions
+
+const events = (currentPage: number) => async (dispatch: AppDispatch) => {
+  try {
+    dispatch(setEventsLoading(true))
+
+    const response = await API.events.getEvents(currentPage)
+
+    console.log(response, 'response')
+    dispatch(setEventsData(response.data.data))
+    dispatch(setPageSize(response.data.pageSize))
+    dispatch(setEventsLoading(false))
+    dispatch(setTotalItems(response.data.totalItems))
```

Comment:
let's add a config to take data from response.data, for rejecting write response.data in each case